nilrt_snac/__main__.py

```diff
@@ -0,0 +1,121 @@
+"""nilrt-snac entry points."""
+
+import argparse
+import logging
+import sys
+from typing import List, Optional
+
+from nilrt_snac._pre_reqs import verify_prereqs
+from nilrt_snac.OpkgHelper import opkg_helper
+from nilrt_snac._configs import CONFIGS
+
+
+from nilrt_snac import Errors, logger, SNACError
+
+
+def _configure(args: argparse.Namespace) -> int:
+    """Configure SNAC mode."""
+    logger.warning("!! THIS TOOL IS IN-DEVELOPMENT AND APPROPRIATE ONLY FOR DEVELOPER TESTING !!")
+    logger.warning("!! Running this tool will irreversibly alter the state of your system.    !!")
+    logger.warning("!! If you are accessing your system using WiFi, you will lose connection. !!")
+
+    if args.yes:
+        consent = "y"
+    else:
+        consent = input("Do you want to continue with SNAC configuration? [y/N] ")
+
+    if consent.lower() not in ["y", "yes"]:
+        return Errors.EX_OK.value
+
+    print("Configuring SNAC mode.")
+    for config in CONFIGS:
+        config.configure(args)
+
+    print("!! A reboot is now required to affect your system configuration. !!")
+    print("!! Login with user 'root' and no password.                       !!")
+
+    return Errors.EX_OK.value
+
+
+def _verify(args: argparse.Namespace) -> int:
+    """Configure SNAC mode."""
+    print("Validating SNAC mode.")
+    valid = True
+    for config in CONFIGS:
+        new_valid = config.verify(args)
+        valid = valid and new_valid
+
+    return Errors.EX_OK if valid else Errors.EX_CHECK_FAILURE
+
+
+def _parse_args(argv: List[str]) -> argparse.Namespace:
+    """Top level entry point for the command line interface."""
+    parser = argparse.ArgumentParser(description="Utility for enabling SNAC mode on NI Linux RT.")
+
+    subparsers = parser.add_subparsers(help="Commands for SNAC mode.", dest="cmd")
+    subparsers.required = True
+
+    configure_parser = subparsers.add_parser("configure", help="Set SNAC mode")
+    configure_parser.add_argument(
+        "-y",
+        "--yes",
+        action="store_true",
+        help="Consent to changes",
+    )
+    configure_parser.set_defaults(func=_configure)
+
+    verify_parser = subparsers.add_parser("verify", help="Verify SNAC mode configured correctly")
+    verify_parser.set_defaults(func=_verify)
+
+    debug_group = parser.add_argument_group("Debug")
+    debug_group.add_argument(
+        "-v",
+        "--verbose",
+        action="store_true",
+        default=0,
+        help="Print debug information.  Can be repeated for more detailed output.",
+    )
+    debug_group.add_argument(
+        "-n",
+        "--dry-run",
+        action="store_true",
+    )
+
+    args = parser.parse_args(argv)
+
+    if 0 < args.verbose and 0 < args.quiet:
```

Comment:
I don't see an `add_parser` for quiet. Is this no longer relevant?

Reply:
It is no longer relevant. Removed.